src/Order/Order.php

```diff
@@ -0,0 +1,960 @@
+            ->withCustomerCounty($json['customerCounty'])
+            ->withCustomerPostcode($json['customerPostcode'])
+            ->withCustomerCountry($json['customerCountry'])
+            ->withCustomerEmail($json['customerEmail'])
+            ->withCustomerTelephone($json['customerTelephone'])
+            ->withCustomerFax($json['customerFax'])
+            ->withCustomerReference($json['customerReference'])
+            ->withCustomerNotes($json['customerNotes'])
+            ->withDeliveryCompany($json['deliveryCompany'])
+            ->withDeliveryName($json['deliveryName'])
+            ->withDeliveryAddress1($json['deliveryAddress1'])
+            ->withDeliveryAddress2($json['deliveryAddress2'])
+            ->withDeliveryAddress3($json['deliveryAddress3'])
+            ->withDeliveryCity($json['deliveryCity'])
+            ->withDeliveryCounty($json['deliveryCounty'])
+            ->withDeliveryPostcode($json['deliveryPostcode'])
+            ->withDeliveryCountry($json['deliveryCountry'])
+            ->withDeliveryTelephone($json['deliveryTelephone'])
+            ->withShippingMethod($json['shippingMethod'])
+            ->withShippingCost($json['shippingCost'])
+            ->withInsurance($json['insurance'])
+            ->withDiscount($json['discount'])
+            ->withVoucherCode($json['voucherCode'])
+            ->withOrderTotal($json['orderTotal'])
+            ->withPaymentComplete($json['paymentComplete'])
+            ->withCurrencyCode($json['currencyCode'])
+            ->withExchangeRate($json['exchangeRate'])
+            ->withSellerUsername($json['sellerUsername'])
+            ->withSellerId($json['sellerId'])
+            ->withCourierProfileName($json['courierProfileName'])
+            ->withBuyerId($json['buyerId'])
+            ->withStoreId($json['storeId'])
+            ->withShipToStore($json['shipToStore'])
+            ->withOrderStatus($json['orderStatus'])
+            ->withFlag1($json['flag1'])
+            ->withFlag2($json['flag2'])
+            ->withCourierName($json['courierName'])
+            ->withCourierService($json['courierService'])
+            ->withCourierServiceCode($json['courierServiceCode'])
+            ->withFulfilmentType($json['fulfilmentType'])
+            ->withWebOrderID($json['webOrderID'])
+            ->withShippingDate($json['shippingDate'])
+            ->withInvoiceDate($json['invoiceDate'])
+            ->withTradeSale($json['tradeSale'])
+            ->withCourierTracking($json['courierTracking'])
+            ->withActualCourierCost($json['actualCourierCost'])
+            ->withOrderItems(
+                ItemSet::fromJson($json['orderItems'] ?? [])
+            )
+            ->withOrderCredits(
+                OrderCreditSet::fromJson($json['orderCredits'] ?? [])
+            )
+            ->withPayments(
+                PaymentSet::fromJson($json['payments'] ?? [])
+            );
+
+    }
+
+    private $orderType;
```

Comment:
Wow bro loads and loads of variables why dont you have a look at https://github.com/snowio/volo-data-model/blob/master/src/OrderUpdate/OrderUpdate.php. It'll show you what you can do to minimize the variable footprint, plus it will promote less bugs. You might need to think  about how it will work for more complex objects like `OrderItemsSet` but its a fun challenge

Reply:
agreed it is too much but I think it is better than having list of properties and validating them when using fromJson toJson to make sure input/output is correct. I think Data Models should be very straightforward and consistent. the idea of using a global array to hold properties just to keep the class short is not really my favorite option. let me know if you have a better solution.